### PR TITLE
Add a new role to provision mobile core on rhpds

### PR DESCRIPTION
**Pull request by wei-lee**
## Motivation

This will allow us to provision mobile core on the OpenShift instances that are created by RHPDS.

To verify:

1. Provision a openshift instance from RHPDS (can use this one: `ssh -i ocp-workshop.pem ec2-user@bastion.mobiletest1.openshiftworkshop.com`, I will email you the private key)
2. SSH into the workstation, and run the following command:
    ```
    sudo bash
    git clone https://github.com/wei-lee/mobile-core.git
    git checkout aerogear-mobile-role
    ansible-playbook installer/rhpds-playbook.yml
    ```
   When it is finished, the mobile tab should be available in the catalog

@johnfriz @maleck13 can you take a look?

### CODE REVIEW

**Comment by wei-lee** [comment]
@evanshortiss FYI. can you try this as well if you have a RHPDS workshop

File: installer/roles/provision-mobile-catalog/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+- name: Copy broker config file
+  copy:
+      src: ../files/broker-config.yaml
+      dest: /tmp/broker-config.yaml
+
+# TODO: ideally we want to parse the broker config to add the new registry and update the config, but for now we keep it simple
+- name: Update ansible service broker config
+  shell: |
+    oc delete configmap broker-config -n {{ansible_service_broker_namespace}}
+    oc create configmap broker-config --from-file=broker-config=/tmp/broker-config.yaml -n {{ansible_service_broker_namespace}}
```
Comment (johnfriz) [inline installer/roles/provision-mobile-catalog/tasks/main.yml]: Is this syntax correct - the two = signs in `--from-file=broker-config=/tmp/broker-config.yaml` seems odd and I can not see anything similar in the docs.

**Comment by evanshortiss** [comment]
@wei-lee worked perfectly for me 👍Will provision a few services and play around with it.

**Comment by evanshortiss** [comment]
`error: update acceptor rejected jenkins-2: pods for rc 'showcase-test/jenkins-2' took longer than 600 seconds to become available` looks like the CI/CD service is struggling to get started, this might be a resourcing issue though since I chose just a single user vs multiple.

**Comment by johnfriz** [comment]
@wei-lee @evanshortiss - Am seeing the same thing. Have tracked it down to the fact that we are requesting a 40GB volume and the rhpds system only provides 10GB volumes (it gives 100 of them, but they are all only 10GB). 

After patching one of the volumes with `oc patch pv/vol1 -p '{"spec":{"capacity":{"storage":"40Gi"}}}'`, I can install the Mobile CI/CD Service. Suggest we add this to the apb so no manual steps are required. Maybe flag it ina . param so it can be set to only run when installing on rhpds. Either that or we drop the pvc to 10Gi - not sure why we would need 40Gi in the first place. 

However, I am still getting an error on provisioning: https://gist.github.com/johnfriz/55f28eb2f476c00b24e11db76c890c2f - have not got to the bottom of this one yet... 

cc @maleck13

File: installer/roles/provision-mobile-catalog/tasks/main.yml (hunk 1 above)
Comment (camilamacedo86) [reply installer/roles/provision-mobile-catalog/tasks/main.yml]: I could found it just here : @https://github.com/openshift-labs/devops-oab-guides/blob/master/ansible/init_prod.yml

**Comment by camilamacedo86** [comment]
Hi @wei-lee,

This PR is opened for a while. Is it something that still required for the installer? Should we move forward here? If yes, could you please update it with the master? If not, could you please close it?

**Comment by wei-lee** [comment]
@witmicko am I right to say that master already supports provisioning on RHPDS?

**Comment by pmacko1** [comment]
@wei-lee just to let you know that it worked for me, same issues with mobile ci/cd though

**Comment by wei-lee** [comment]
@pmacko1 thanks. Did you mean master branch?

**Comment by pmacko1** [comment]
@wei-lee no, i checked out `aerogear-mobile-role`.

**Comment by camilamacedo86** [comment]
@wei-lee it needs to be rebased with the master? Could you do it? Also, please let us know if it still an required change

**Comment by wei-lee** [comment]
@camilamacedo86 I am not sure if this PR is still required, as I think @witmicko might have these changes in master via another PR. I am just waiting for @witmicko to confirm. 

@witmicko can you confirm if the master is already capable of provision on rhpds?

**Comment by pmacko1** [comment]
Hi @wei-lee, will this eventually work for the new release as well (data sync)? We are using rhpds for docs review. Thanks !

**Comment by pmacko1** [comment]
bump.

is this still valid? docs team have problems setting the cluster up locally, so this is quite ideal for verification. Will mdc and data sync support this? @wei-lee @maleck13 @camilamacedo86

**Comment by wei-lee** [comment]
This project is deprecated. Closing the PR.

**Pull request closed without merging**